src/PetrolCar.cpp

```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+
+#include "PetrolCar.hpp"
+
+PetrolCar::PetrolCar(std::unique_ptr<PetrolEngine> engine)
+    : engine_(std::move(engine)) {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+PetrolCar::~PetrolCar() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void PetrolCar::refuel() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void PetrolCar::restore() {
+    refuel();
+}
+
+void PetrolCar::changeEngine(std::unique_ptr<PetrolEngine> engine) {
+    if (getSpeed() != 0) {
+        throw InvalidEngineChange("Can't change engine while it is running");
+    }
+    engine_ = std::move(engine);
```

Comment:
memory leak.

Reply:
@ziobron I have checked this on valgrind. Below it's result
==5824==
==5824== HEAP SUMMARY:
==5824== in use at exit: 0 bytes in 0 blocks
==5824== total heap usage: 16 allocs, 16 frees, 74,440 bytes allocated
==5824==
==5824== All heap blocks were freed -- no leaks are possible
==5824==
==5824== For lists of detected and suppressed errors, rerun with: -s
==5824== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)